### PR TITLE
feat(tab-bar): typescript

### PR DESCRIPTION
**Pull request by moog16**
related: #48

### CODE REVIEW

File: packages/tab-bar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+export interface TabBarProps extends React.HTMLProps<HTMLDivElement> {
+  indexInView?: number;
+  activeIndex: number;
+  handleActiveIndexUpdate?: (index: number) => void;
+  className?: string;
+  isRtl?: boolean;
+};
+
+interface TabBarState {
+  previousActiveIndex: any;
```
Comment (patrickrodee) [inline packages/tab-bar/index.tsx]: Can we make this more strict than `any`?

File: packages/tab-bar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+};
+
+export default class TabBar extends React.Component<
+  TabBarProps, TabBarState
+  > {
+  tabBarElement_: React.RefObject<HTMLDivElement> = React.createRef();
+  tabScroller_: React.RefObject<TabScroller> = React.createRef();
+  tabList_: Tab[] = [];
+  foundation_: MDCTabBarFoundation | null = null;
```
Comment (patrickrodee) [inline packages/tab-bar/index.tsx]: Can this be optional instead of nullable?

File: packages/tab-bar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+};
+
+interface TabBarState {
+  previousActiveIndex: any;
+};
+
+export default class TabBar extends React.Component<
+  TabBarProps, TabBarState
+  > {
+  tabBarElement_: React.RefObject<HTMLDivElement> = React.createRef();
```
Comment (patrickrodee) [inline packages/tab-bar/index.tsx]: Can these be private?

File: packages/tab-bar/index.tsx (hunk 1 above)
Comment (moog16) [reply packages/tab-bar/index.tsx]: ah good catch

File: packages/tab-bar/index.tsx (hunk 2 above)
Comment (moog16) [reply packages/tab-bar/index.tsx]: certainly!

File: packages/tab-bar/index.tsx (hunk 3 above)
Comment (moog16) [reply packages/tab-bar/index.tsx]: This is also referenced in `/test/unit/tab-bar/index.test.tsx`.

**Pull request merged**